eng/pipelines/templates/jobs/live.tests.yml

```diff
@@ -78,6 +78,9 @@ jobs:
       - template: ../steps/common.yml
 
       - ${{ if ne(parameters.ServiceDirectory, '') }}:
+
```

Comment:
Moved here so that we can run presteps that apply to test resource deployment but I am not sure if there's a better place? @benbp what do you think?

Matches dotnet to an extent https://github.com/Azure/azure-sdk-for-net/blob/main/eng/pipelines/templates/jobs/live.tests.yml#L103

Reply:
cc @ckairen @mikeharder @weshaggard if y'all have a recommendation (no rush)

btw, I checked and the only other usage I see is related to cosmos and setting up an emulator. [search results](https://github.com/search?q=repo%3AAzure%2Fazure-sdk-for-js+presteps&type=code) 

but, still curious if you have feedback and/or if I missed anything obvious?

Reply:
It looks like the one other place that `PreSteps` is used (https://github.com/Azure/azure-sdk-for-js/blob/main/eng/pipelines/templates/stages/cosmos-sdk-client.yml#L45-L46) has an empty `ServiceDirectory` parameter, meaning they will be skipped with this update. I would put the `PreSteps` above the `- ${{ if ne(parameters.ServiceDirectory, '') }}:` conditional so it always runs.

Reply:
Otherwise this looks good, pre steps should have been before the deploy step anyway I think.

Reply:
Moved pre-steps above the conditional 👍 thanks!